### PR TITLE
irc: add support for stateless bridging via draft/relaymsg

### PR DESCRIPTION
**Pull request by jlu5**
As discussed at https://github.com/42wim/matterbridge/issues/667#issuecomment-634214165

### CODE REVIEW

**Comment by poVoq** [comment]
This is meant to work with the Oragono IRC server, right?

**Comment by jlu5** [comment]
@poVoq This will work with Oragono as well as InspIRCd with a [contrib module](https://github.com/inspircd/inspircd-contrib/blob/master/3.0/m_relaymsg.cpp). (Of course, other server implementations would be welcome too)

File: bridge/irc/irc.go [hunk 1]
```diff
@@ -205,27 +205,57 @@ func (b *Birc) doConnect() {
...
 	for msg := range b.Local {
 		<-throttle.C
 		username := msg.Username
-		if b.GetBool("Colornicks") && len(username) > 1 {
-			checksum := crc32.ChecksumIEEE([]byte(msg.Username))
-			colorCode := checksum%14 + 2 // quick fix - prevent white or black color codes
-			username = fmt.Sprintf("\x03%02d%s\x0F", colorCode, msg.Username)
-		}
+		// Optional support for the proposed RELAYMSG extension, described at
+		// https://github.com/jlu5/ircv3-specifications/blob/master/extensions/relaymsg.md
```
Comment (42wim) [inline bridge/irc/irc.go:226]:
```suggestion
		// https://github.com/jlu5/ircv3-specifications/blob/master/extensions/relaymsg.md
		// nolint:nestif
```

Disable the nestif linter here, I'll refactor later.

File: bridge/irc/irc.go [hunk 2]
```diff
@@ -205,27 +205,57 @@ func (b *Birc) doConnect() {
...
-		case config.EventUserAction:
-			b.i.Cmd.Action(msg.Channel, username+msg.Text)
-		case config.EventNoticeIRC:
-			b.Log.Debugf("Sending notice to channel %s", msg.Channel)
-			b.i.Cmd.Notice(msg.Channel, username+msg.Text)
-		default:
-			b.Log.Debugf("Sending to channel %s", msg.Channel)
-			b.i.Cmd.Message(msg.Channel, username+msg.Text)
+			if msg.Event == config.EventUserAction {
+				b.i.Cmd.SendRawf("RELAYMSG %s %s :\x01ACTION %s\x01", msg.Channel, username, text)
```
Comment (42wim) [inline bridge/irc/irc.go:238]:
```suggestion
				b.i.Cmd.SendRawf("RELAYMSG %s %s :\x01ACTION %s\x01", msg.Channel, username, text) //nolint:errcheck
```

Disable errcheck here

File: bridge/irc/irc.go [hunk 3]
```diff
@@ -205,27 +205,57 @@ func (b *Birc) doConnect() {
...
+			} else {
+				b.Log.Debugf("Sending RELAYMSG to channel %s: nick=%s", msg.Channel, username)
+				b.i.Cmd.SendRawf("RELAYMSG %s %s :%s", msg.Channel, username, text)
```
Comment (42wim) [inline bridge/irc/irc.go:241]:
```suggestion
				b.i.Cmd.SendRawf("RELAYMSG %s %s :%s", msg.Channel, username, text) //nolint:errcheck
```

**Review by 42wim — COMMENTED** [review]
You can add the suggested //nolint's to disable the error checking, but please run the `gofmt -s` as suggested by the linter.

**Comment by 42wim** [comment]
Thanks for this PR!
Could you also include `UseRelayMsg` in the `matterbridge.toml.sample` file?

**Comment by 42wim** [comment]
Thanks!

**Pull request merged**